### ISSUE TITLE
Deprecate add_pool/K for K > 2

### ISSUE THREAD

**Issue by jlouis** [Eonblast/Emysql#111]
The `add_pool/K` calls are a mess to work with operationally. They should be deprecated:
- Improve error checking for `add_pool/2` so it will bail if something is incorrectly given.
- Map all the `add_pool/K` variants to `add_pool/2`
- Plan a release where the calls have been removed.

**Comment by hdiedrich** [comment]
+1 it would be cool to have a way to ask if anyone actually uses a feature.

**Comment by jlouis** [comment]
It is somewhat hard. You can add

```
-deprecated({add_pool, 8}).
```

tags and `xref` can read those and warn users. But this means projects must use xref, and few projects do, unfortunately. So my plan is to rebuild the interface around a `/2` variant and then steer every other variant onto the path of this one.

**Comment by hdiedrich** [comment]
Maybe a low frequency mailing list, or twitter feed would be warranted for such announcements.

**Comment by saa** [comment]
@hdiedrich This is not the best way. Not all follow the mailing lists and twitter.

**Comment by evabihari** [comment]
add_pool/2 spec should be updated - with the actual implementation Result can be the following: 
Result = {reply, {error, pool_already_exists}, state()} | {reply, ok, state() | {error, pool_already_exists}}

**Comment by jlouis** [comment]
Good catch! @evabihari - we should definitely update the spec so it matches the expected values.
